File: LabProgII/Python/Battleship/server/jogador.py
class Jogador:
    def __init__(self, id, addr, conn):
        self.id = id
        self.addr = addr
        self.conn = conn
        self.campoJogador = gerar_matriz(10,10)
        self.campoAdversario = gerar_matriz(10,10)
        self.totalAbatidos = 0
        self.totalRestante = 10

    def atingirJogador(self, x,y):
        if(self.campoJogador[x][y]=='1'):
            self.totalAbatidos+=1
            self.totalRestante-=1
            self.campoJogador[x][y]='X'
            return True
        self.campoJogador[x][y]='~'
        return False
    
    def atingirAdversario(self,x,y,ans):
        if(ans):
            self.campoAdversario[x][y]='X'
            return
        self.campoAdversario[x][y]='~'

def gerar_matriz (n_linhas, n_colunas):
    return [[" "]*n_colunas for _ in range(n_linhas)]

File: LabProgII/Java/BatalhaNaval/src/Game/Jogador.java
package Game;

public class Jogador {
	int id;
	char[][] campoJogador = new char[10][10]; //visao do jogador
	char[][] campoAdversario = new char[10][10]; //visao do adversario
	int totalAbatidos = 0;
	int totalRestante = 3;
	
	public Boolean atingirJogador(int linha, int coluna) {
		if(campoJogador[linha][coluna]=='1') {
			this.totalAbatidos++;
			this.totalRestante--;
			campoJogador[linha][coluna]='X';
			return true;
		} 
		campoJogador[linha][coluna]='~';
		return false;
	}
	
	public void atingirAdversario(int linha, int coluna, Boolean ans) {
		if(ans) {
			campoAdversario[linha][coluna]='X';
			return;
		} 
		campoAdversario[linha][coluna]='~';
	}
	
	public Boolean posicionarNavio(int linha, int coluna) {
		if (this.campoJogador[linha][coluna] == '1') {
			System.out.printf("--> Posição ocupada, tente novamente!\n");
			return false;
		}
		this.campoJogador[linha][coluna]='1';
		System.out.printf("--> Barco adicionado em (%d,%d)\n",linha,coluna);
		return true;		
	}
}